### PR TITLE
Role based access control

### PR DESCRIPTION
**Pull request by Fontie**
JWT token gets saved, get pushed back to login when not logged in, initial nav bar that blocks depending on rights.

### CODE REVIEW

**Review by MilanBrock — DISMISSED** [review]
i trust fontie 😎

**Review by Popodil — DISMISSED** [review]

File: src/router/index.ts [hunk 1]
```diff
@@ -30,28 +64,65 @@ const router = createRouter({
...
+      next({ name: 'login' });
+    } else {
+      const decoded = decodeJWT(token);
+      const roles = decoded.realm_access?.roles || [];
+      console.log(roles);
+
+      // Role-based access check
+      if (to.meta.allowedRoles && !to.meta.allowedRoles.some(role => roles.includes(role))) 
+      {
+        console.log("I TRIED TO STOP HIM I SWEAR!!!");
```
Comment (Popodil) [inline src/router/index.ts:116]: Console.log could go

**Review by Popodil — APPROVED** [review]

File: src/components/Navbar.vue [hunk 2]
```diff
@@ -14,26 +11,104 @@ import Form from "./Form.vue";
...
+      // Decode the JWT token and parse the payload
+      const payload = JSON.parse(atob(token.split('.')[1]));
+      roles.value = payload.realm_access?.roles || [];
+    } catch (error) {
+      console.error('Error decoding token:', error);
+    }
+  }
+  else
+  {
+    console.log("no role");
```
Comment (MilanBrock) [inline src/components/Navbar.vue:101]: Should remember to check all logs for final release

File: src/router/index.ts (hunk 1 above)
Comment (MilanBrock) [inline src/router/index.ts:116]: Please remove un-serious logs in the future

File: src/views/Login.vue [hunk 3]
```diff
@@ -43,31 +43,41 @@ import axios from 'axios';
...
-        if (response.status == 200) {
-          localStorage.setItem('token', response.data.token);
-          this.$router.push('/LobbyCreate');
-        } else 
-        {
-          this.errorMessage = 'Invalid email or password.';
+        if (response.status === 200) {
+          // Store the token in localStorage
+          const token = response.data.startsWith("Bearer ") ? response.data.split(" ")[1] : response.data;
+          console.log(token);
```
Comment (Popodil) [inline src/views/Login.vue:73]: Do you need this??

**Review by MilanBrock — APPROVED** [review]
Approved!

**Pull request merged**